public/app/core/services/__mocks__/search_srv.ts

```diff
@@ -0,0 +1,22 @@
+export const mockSearch = jest.fn<any, any>(() => {
+  return Promise.resolve([]);
+});
+
+export const mockGetDashboardTags = jest.fn(async () =>
+  Promise.resolve([
+    { term: 'tag1', count: 2 },
+    { term: 'tag2', count: 10 },
+  ])
+);
+
+export const mockGetSortOptions = jest.fn(() =>
+  Promise.resolve({ sortOptions: [{ name: 'test', displayName: 'Test' }] })
+);
+
+export const SearchSrv = jest.fn().mockImplementation(() => {
+  return {
+    search: mockSearch,
+    getDashboardTags: mockGetDashboardTags,
+    getSortOptions: mockGetSortOptions,
+  };
+});
```

Comment:
I think this ok for this PR but I find this convention using files under `__mocks__` very confusing when you try to debug/change test you haven't written. I remember one time me and @mckn spent like 30 minutes trying to understand a test that didn't behave like we understood it would because of a file under `__mocks__`

I'd like for us to have a discussion about this during a future FrontEnd weekly to hear what the consensus is amongst everyone.

Reply:
@hugohaggmark thanks for the feedback, wasn't aware that this causes some difficulties. Just to clarify, do you mean that we shouldn't use automatic mocks? I'm wondering what would be the alternative? In this case the mock is not applied unless `jest.mock('app/core/services/search_srv');` is called in the test file, so that in my opinion should be a clear indicator that the mock functionality is used.

Reply:
Yes, I agree with you, that should be a sign. But most of our tests include the mock within the same test file, so before I didn't pay so much attention to that line. Now that I've been burned by this I pay more attention to this line.

My personal opinion is that it should be crystal clear in each test what mocks/dependencies you're using and we should define them in the same file. I'm probably the odd bird here so a proper discussion on the way forward would be great :)

Reply:
I see your point and I guess we do need to have some discussion as I was just thinking that actually having mocks in `__mock__` files is an improvement that allows reusing them without mocking the functionality in each test file :)